### PR TITLE
Fix debug print to support removed inputs

### PR DESCRIPTION
**Pull request by unnonouno**
Fix `DebugPrint` extension to support removed inputs. I also fixed its test to check print result.
related to #2452

### CODE REVIEW

**Comment by beam2d** [comment]
LGTM.

**Comment by beam2d** [comment]
The tests on Win+Py2 do not pass, maybe due to the shape printed as `(3L, 5L)`. Is it ok?

File: tests/chainer_tests/function_hooks_tests/test_debug_print.py [hunk 1]
```diff
@@ -1,72 +1,138 @@
...
     def test_forward_cpu(self):
-        self.f(chainer.Variable(self.x))
+        self.f.add_hook(self.h)
+        self.f(chainer.Variable(self.x), chainer.Variable(self.x))
+        expect = '''^function\tDummyFunction
+input data
+<variable at 0x[0-9a-f]+>
+- device: CPU
+- backend: <(type|class) 'numpy.ndarray'>
+- shape: \(3, 5\)
```
Comment (beam2d) [inline tests/chainer_tests/function_hooks_tests/test_debug_print.py]: The tests on Win+Py2.7 do not pass, maybe due to this line (it is printed as `(3L, 5L)` in this environment). Can you fix it? It's enough to just fix these test cases (including this line and others below).

Comment (unnonouno) [inline tests/chainer_tests/function_hooks_tests/test_debug_print.py]: Oh, OK

**Comment by beam2d** [comment]
LGTM

**Pull request merged**